plugins/jobs/src/js/JobCreateEditFormModal.tsx

```diff
@@ -0,0 +1,103 @@
+// tslint:disable:no-empty
+import * as React from "react";
+import { Observable } from "rxjs/Observable";
+import { BehaviorSubject } from "rxjs/BehaviorSubject";
+import { combineLatest } from "rxjs/observable/combineLatest";
+import "rxjs/add/operator/switchMap";
+
+import { componentFromStream } from "data-service";
+
+import MetronomeStore from "#SRC/js/stores/MetronomeStore";
+import Job from "#SRC/js/structs/Job";
+import {
+  METRONOME_JOB_CREATE_ERROR,
+  METRONOME_JOB_CREATE_SUCCESS,
+  METRONOME_JOB_UPDATE_ERROR,
+  METRONOME_JOB_UPDATE_SUCCESS
+} from "#SRC/js/constants/EventTypes";
+
+import JobFormModal, { ErrorMessage } from "./components/JobFormModal";
+
+interface JobCreateEditFormModalProps {
+  isEdit?: boolean;
+  job?: Job;
+  open: boolean;
+  onClose: () => void;
+}
+
+type IsJsonMode = boolean;
+
+type JobFormModalInput = [
+  JobCreateEditFormModalProps,
+  IsJsonMode,
+  ErrorMessage
+];
+
+const JobCreateEditFormModal = componentFromStream<JobCreateEditFormModalProps>(
+  props$ => {
+    const isJsonMode$ = new BehaviorSubject<IsJsonMode>(false);
+    const errorMessage$ = new BehaviorSubject<ErrorMessage>(null);
+
+    function handleInputModeToggle() {
+      isJsonMode$.next(!isJsonMode$.value);
```

Comment:
Hmmm, is this WIP, or does this actually work?

Reply:
It _does_ work :D though it doesn't close the modal on success. Plus I still don't like it.

Reply:
sure, for this you have to call `props.onClose` - but maybe `props$.value.onClose` works? 🤔